presto-spi/src/main/java/com/facebook/presto/spi/statistics/ConnectorHistogram.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * This interface contains functions which the Presto optimizer can use to
+ * answer questions about a particular column's data distribution. These
+ * functions will be used to return answers to the query optimizer to build
+ * more realistic cost models for joins and filter predicates.
+ * <br>
+ * Currently, this interface supports representing histograms of columns whose
+ * domains map to real values.
+ * <br>
+ * Null values should not be represented in underlying histogram implementation.
+ * When calculating filter statistics using the {@link ColumnStatisticType#NUMBER_OF_NON_NULL_VALUES}
+ * are used to account for nulls in cost-based calculations.
+ *
+ * @see ColumnStatisticType#NUMBER_OF_NON_NULL_VALUES
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.MINIMAL_CLASS, property = "@class")
+public interface ConnectorHistogram
```

Comment:
Are there examples of connector specific histograms. I'm curious about the reasons why different connectors might have distinct, incompatible histograms?

Reply:
Hive 4 is planning to support storage of Apache Datasketches KLL sketches. Spark has its own standard binned histogram representation that is stored in a Base64-encoded binary format on the table properties. Presto also supports creating T-Digests through `tdigest_agg`. Iceberg could support any of these histogram representations in its Puffin file format. There are numerous histograms implementations as well as supported storage locations usually dependent upon the connector. I want to support reading or writing any type of histogram from any type of storage.

The actual implementation of any histogram is not included in this PR though. It is just the framework for the optimizer to make use of them.